src/app/components/coupon-profile/coupon-profile.component.ts

```diff
@@ -26,25 +26,12 @@ export class CouponProfileComponent implements OnInit {
 
   ngOnInit(): void {
     this.activatedRoute.params.subscribe((params) => {
-      this.getModel(params.id).subscribe((value: Coupon) => {
+      this.generalService.getCoupon(params.id).subscribe((value: Coupon) => {
         this.coupon = value;
       });
     });
   }
 
-  getModel(id: number): Observable<Coupon> {
-    if (this.user) {
-      switch (this.user.clientType) {
-        case ClientType.COMPANY:
-          return this.companyService.getCouponById(id);
-        case ClientType.CUSTOMER:
-          return this.customerService.getCoupon(id);
-      }
-    }
-    return this.generalService.getCoupon(id);
-
-  }
-
```

Comment:
so in here you used generalService.getCoupon instead, why did you decide that getModel is not necessary? is it because you wanted the model to come from the general service and not locally?